batch_requests.go

```diff
@@ -0,0 +1,150 @@
+package msgraphgocore
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/google/uuid"
+	abstractions "github.com/microsoft/kiota-abstractions-go"
+	"github.com/microsoftgraph/msgraph-sdk-go-core/internal"
+)
+
+// SendBatch sends a batch request
+func SendBatch(batch batchRequest, adapter abstractions.RequestAdapter) (BatchResponse, error) {
+	var res BatchResponse
+
```

Comment:
```suggestion
```

Reply:
res is used in the error checks.

Reply:
sure but it's never initialized, you could return nil instead (would require changing the return type to a pointer)

Reply:
Go initialises it with zero values of the struct fields. There's a cost to returning pointers which doesn't make sense to pay in this scenario. Pointers to structs means that the struct is stored in the heap instead of stack.

Reply:
how is a caller supposed to tell the difference between no response but normal execution, no response but failed execution, or response and successful execution? wouldn't nil/nil, nil/err, value/nil be easier to map for them? That's the pattern we have throughout the SDK so far.

Reply:
Go by convention checks for failed execution depending on whether the returned err is nil or not.

Reply:
sure, and in the current service implementation there's always going to be a response value unless there's an error. resolving.